docs/source/backend.rst

```diff
@@ -7,7 +7,7 @@ torchaudio.backend
 To switch backend, use :py:func:`torchaudio.set_audio_backend`. To check the current backend use :py:func:`torchaudio.get_audio_backend`.
 
 .. warning::
-   Although ``sox`` backend is default for backward compatibility reason, it has a number of issues, therefore it is highly recommended to use ``sox_io`` backend instead. Note, however, that due to the interface refinement, functions defined in ``sox`` backend and those defined in ``sox_io`` backend do not have the signatures.
+   Although ``sox`` backend is default for backward compatibility reason, it has a number of issues, therefore it is highly recommended to use ``sox_io`` backend instead. Note, however, that due to the interface refinement, functions defined in ``sox`` backend and those defined in ``sox_io`` backend do not have the same signatures.
```

Comment:
Actually,  let's simplify the statement:
```
``sox`` backend is currently the default for backward compatibility reason. It is recommended to migrate to ``sox_io`` backend, and use its new interface, which will become the default in a future version. See [here](link) for more information about how to migrate.
```

Reply:
It's better to do that after we figure out the plan for deprecation of `sox` backend.

Reply:
My point is mostly to remove the idea of "issues", "interface refinement". How about this then?
```
``sox`` backend is the default for backward compatibility reason, but it is recommended to use ``sox_io`` backend instead. Note, however, functions defined in ``sox`` backend and those defined in ``sox_io`` backend do not have the same signatures.
```

Although ``sox`` backend is default for backward compatibility reason, it has a number of issues, therefore it is highly recommended to use ``sox_io`` backend instead. Note, however, that due to the interface refinement, functions defined in ``sox`` backend and those defined in ``sox_io`` backend do not have the signatures.

Reply:
My point here is to discourage users to use `sox` backend and raise the awareness of the danger of using it. If we are not upfront of issues, users would not take efforts to migrate because it takes engineering efforts to do so.

Reply:
Can you elaborate on "dangers"?

Reply:
As it's summarized in https://github.com/pytorch/audio/pull/726

`sox` backend does not load wav files other than 16-bit signed integer, `save` function does not preserve data, `info` returns the number of samples instead of the number of frames. This affects every aspect of ML application, from training to inference. There is always chance to introduce wrong data when using `load` and if one works on generative models, `save` function does not correctly save the data.

Ideally we should fix it but the inconsistent design and band aides placed in a wrong place to fix the issue(load_wav), it's very difficult to do that now without breaking BC.

Reply:
I understand though I don't think this is the correct wording. Since I do not want to block this change here for this, I'll go ahead and approve this pull request anyway.

Reply:
Thanks for the clarification. I understand what you mean though I don't think this is the correct wording. Since I do not want to block this change here for this, I'll approve the pull request.